package/MDAnalysis/analysis/base.py

```diff
@@ -652,3 +654,236 @@ def _filter_baseanalysis_kwargs(function, kwargs):
+        """
+        if len(objects) == 1:
+            rv = objects[0]
+        else:
+            rv = Results()
+            for key in objects[0].keys():
+                agg_function = self._lookup.get(key, None)
+                if agg_function is None and do_raise:
+                    raise ValueError(f"No aggregation function for {key=}")
+                results_of_t = [obj[key] for obj in objects]
+                rv[key] = agg_function(results_of_t)
+        return rv
+
+    @staticmethod
+    def flatten_sequence(arrs: list[list]):
+        return [item for sublist in arrs for item in sublist]
+
+    @staticmethod
+    def ndarray_sum(arrs: np.ndarray):
+        return np.array(arrs).sum(axis=0)
+
+    @staticmethod
+    def ndarray_mean(arrs: np.ndarray):
+        return np.array(arrs).sum(axis=0)
+
+    @staticmethod
+    def float_mean(floats: list[float]):
+        return sum(floats) / len(floats)
+
+
+class ParallelAnalysisBase(AnalysisBase):
+
+    def __init__(self, trajectory, verbose=False, **kwargs):
+        super().__init__(trajectory, verbose, **kwargs)
+
+    def _setup_computation_groups(
+        self, n_parts: int, start=None, stop=None, step=None, frames=None
+    ) -> list[np.ndarray]:
+        """
+        Splits the trajectory frames, defined by `start/stop/step` or `frames`, into
+        `n_parts` even groups, preserving their indices.
+
+        Parameters
+        ----------
+        n_parts : int
+            number of parts to split the workload into
+        start : int, optional
+            start frame
+        stop : int, optional
+            stop frame
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+
+        Raises
+        ------
+        ValueError
+            if *both* `frames` and at least one of `start`, `stop`, or `frames`
+            is provided (i.e., set to another value than ``None``)
+
+        Returns
+        -------
+        computation_groups : list[np.ndarray]
+            list of (n, 2) shaped np.ndarrays with frame indices and numbers
+        """
+        if frames is None:
+            start, stop, step = self._trajectory.check_slice_indices(start, stop, step)
+            used_frames = np.arange(start, stop, step)
+        elif not all(opt is None for opt in [start, stop, step]):
+            raise ValueError("start/stop/step cannot be combined with frames")
+        else:
+            used_frames = frames
+
+        if all((isinstance(obj, bool) for obj in used_frames)):
+            arange = np.arange(len(used_frames))
+            used_frames = arange[used_frames]
+
```

Comment:
This looks like code that's duplicated in `AnalysisBase.run()` and would eventually be refactored.